grails-app/controllers/org/pih/warehouse/api/PartialReceivingApiController.groovy

```diff
@@ -57,6 +60,106 @@ class PartialReceivingApiController {
+            partialReceipt?.partialReceiptContainers?.partialReceiptItems?.add(new PartialReceiptItem())
+        }
+
+        def lineItems = partialReceipt.partialReceiptItems.collect {
+            [
+            "Receipt item id": it?.receiptItem?.id ?: "",
+            "Shipment item id": it?.shipmentItem?.id ?: "",
+            Code: it?.shipmentItem?.product?.productCode ?: "",
+            Name: it?.shipmentItem?.product?.name ?: "",
+            "Lot/Serial No.": it?.lotNumber ?: "",
+            "Expiration date": it?.expirationDate?.format("MM/dd/yyyy") ?: "",
+            "Bin Location": it?.binLocation ?: "",
+            Recipient: it?.recipient?.id ?: "",
+            Shipped: it?.quantityShipped ?: "",
+            Received: it?.quantityReceived ?: "",
+            "To receive": it?.quantityRemaining ?: "",
+            "Receiving now": it?.quantityReceiving ?: "",
+            Comment: it?.comment ?: ""
+            ]
+        }
+
+        String csv = dataService.generateCsv(lineItems)
+        response.setHeader("Content-disposition", "attachment; filename=\"PartialReceiving-${params.id}.csv\"")
+        render(contentType:"text/csv", text: csv.toString(), encoding:"UTF-8")
+    }
+
+    def importCsv = { ImportDataCommand command ->
+
+        try {
+            PartialReceipt partialReceipt = receiptService.getPartialReceipt(params.id, "1")
+
+            def importFile = command.importFile
+            if (importFile.isEmpty()) {
+                throw new IllegalArgumentException("File cannot be empty")
+            }
+
+            if (importFile.fileItem.contentType != "text/csv") {
+                throw new IllegalArgumentException("File must be in CSV format")
+            }
+
+            String csv = new String(importFile.bytes)
+            def settings = [separatorChar: ',', skipLines: 1]
+            csv.toCsvReader(settings).eachLine { tokens ->
+                String receiptItemId = tokens[0] ?: null
+                String shipmentItemId = tokens[1] ?: null
+                String recipientId = tokens[7] ?: null
+                Integer quantityReceiving = tokens[11] ? tokens[11].toInteger() : null
+                String comment = tokens[12] ? tokens[12] : null
+
+                List<PartialReceiptItem> partialReceiptItems = []
+                partialReceipt.partialReceiptItems.each {
+                    partialReceiptItems.addAll(it)
+                }
+
+                PartialReceiptItem partialReceiptItem = partialReceiptItems.find { receiptItemId ? it?.receiptItem?.id == receiptItemId : it?.shipmentItem?.id == shipmentItemId }
+
+                if (!partialReceiptItem) {
+                    throw new IllegalArgumentException("Receipt item id: ${receiptItemId} not found")
+                }
+
+                Person recipient = recipientId ? Person.get(recipientId) : null
+                if (!recipient && recipientId) {
+                    String[] names = recipientId.split(" ")
```

Comment:
if you export recipient id I think you should not try to split it as a name
you're not using the recipient anywhere, so just remove this for now